packages/widgets/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,48 @@
+import styled from '@emotion/styled';
+import { usePaginationRange } from '@ssa-ui-kit/hooks';
+
+import { ArrowButton } from './ArrowButton';
+import { PaginationButtons } from './PaginationButtons';
+
+import { IPaginationProps } from './types';
+import { usePaginationContext } from './PaginationContext';
+
+const Nav = styled.nav``;
+
+const Pagination = ({
+  pagesCount,
+  className,
+  as,
+  ariaLabel,
+}: IPaginationProps) => {
+  const { page, setPage } = usePaginationContext();
+  const range = usePaginationRange({ pagesCount, selectedPage: page });
+
+  return (
+    <Nav className={className} as={as} aria-label={ariaLabel || 'Pagination'}>
+      <ArrowButton
+        direction="left"
+        onClick={() => {
+          if (page && page > 1) {
+            setPage(page - 1);
+          }
+        }}
+        isDisabled={page === 1}
+        css={{ marginRight: '12px' }}
+      />
+      <PaginationButtons range={range} selectedPage={page} onClick={setPage} />
+      <ArrowButton
+        direction="right"
+        onClick={() => {
+          if (page && page < pagesCount) {
```

Comment:
Maybe, we need to add some new rules for the `isDisabled` property? Otherwise, we can have the situation when the button is active, user is clicked it, but nothing is happened

Reply:
> Otherwise, we can have the situation when the button is active, user is clicked it, but nothing is happened. 

Can you give an example of such a case? 

Currently, when the 1st or the last page is selected, the "Previous page button" or the "Next page button" respectively become disabled. So, no click event happens. Also, if there's no page selected or only one page selected, the click event will happen, but the `setPage()` won't be called. 

We probably need to disable these buttons when no pages or one page exists. I think that can be a good adjustment to the existing logic. What do you think?

Reply:
I understood you point of view. In this case, I think we have redundant comparison  `page > 1` and `page < pagesCount`, owing to `isDisabled` property. If it's needed, we can keep in place `if (page) {...}` comparison, but I think we can also pass the certain value of page and avoid this comparison also.

And yes, I think it's a good approach to disable these buttons when no pages or one page exists.